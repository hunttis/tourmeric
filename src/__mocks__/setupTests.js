import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16'; // eslint-disable-line

configure({ adapter: new Adapter() });
