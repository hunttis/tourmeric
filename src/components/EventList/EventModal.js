import React, { Component, Fragment } from 'react';
import { Translate } from 'react-localize-redux';
import PropTypes from 'prop-types';
import _ from 'lodash';
import Moment from 'react-moment';
import { isLoaded } from 'react-redux-firebase';

export default class EventModal extends Component {

  foo() {

  }

  render() {
    const { eventId, events, settings, closeModal, participations, categories } = this.props;

    if (isLoaded(events) && isLoaded(participations) && isLoaded(settings)) {

      const eventContent = _.get(events, eventId, {});
      const category = _.get(categories, eventContent.category);
      const dateFormat = _.get(settings, 'dateFormat');
      let participationsForEvent = Object.values(_.get(participations, eventId, []));
      participationsForEvent = _.sortBy(participationsForEvent, ['date']);

      return (
        <div className="modal is-active" id={`modal${eventId}`} key={`modal${eventId}`}>
          <div className="modal-background" onClick={closeModal} />
          <div className="modal-content">
            <div className="box is-rounded">
              <div className="columns is-multiline">

                <div className="column is-9">
                  <h1 className="title has-text-success">{eventContent.name}</h1>
                </div>

                {category &&
                <div className="column is-3 has-text-right">
                  <figure className="image is-64x64 is-pulled-right">
                    <img alt="" src={category.image} />
                  </figure>
                </div>
            }

                <div className="column is-12">
                  {eventContent.date &&
                  <Fragment>
                    <i className="fas fa-calendar" />&nbsp;&nbsp;
                    <Moment format={dateFormat}>{eventContent.date}</Moment>
                    <br />
                  </Fragment>
              }

                  {eventContent.time &&
                  <Fragment><i className="fas fa-clock" />&nbsp;&nbsp;{eventContent.time}<br /></Fragment>
              }

                  {eventContent.format &&
                  <Fragment><i className="fas fa-book" />&nbsp;&nbsp;{eventContent.format}<br /></Fragment>
              }

                  {eventContent.rulesLevel &&
                  <Fragment><i className="fas fa-balance-scale" />&nbsp;&nbsp;{eventContent.rulesLevel}<br /></Fragment>
              }

                  {eventContent.entryFee &&
                  <Fragment><i className="fas fa-money-bill-alt" />&nbsp;&nbsp;{eventContent.entryFee}&nbsp;â‚¬<br /></Fragment>
              }
                </div>

                <ModalItem translationKey="notes" content={eventContent.notes} />
                <ModalItem translationKey="prizes" content={eventContent.prizes} />

                {eventContent.link &&
                <Fragment>
                  <div className="column is-12">
                    <div className="subtitle has-text-info"><Translate id="link" /></div>
                  </div>

                  <div className="column is-1" />
                  <div className="column is-11">
                    <p>
                      <a href={eventContent.link}>{eventContent.link}</a>
                    </p>
                  </div>
                </Fragment>
            }
                {!_.isEmpty(participations) &&
                <Fragment>

                  <div className="column is-12">
                    <div className="subtitle has-text-info"><Translate id="participants" /> ({participationsForEvent.length} / {eventContent.playerSlots})</div>
                  </div>
                  <div className="column is-1" />
                  <div className="column is-11">
                    <ParticipantList participations={participationsForEvent} />
                  </div>
                </Fragment>
            }
              </div>
            </div>
          </div>
          <button className="modal-close is-large" aria-label="close" onClick={closeModal} />
        </div>
      );
    }
    return <div />;
  }
}

const ModalItem = ({ translationKey, content }) => (
  <Fragment>
    {content &&
    <Fragment>
      <div className="column is-12">
        <div className="subtitle has-text-info"><Translate id={translationKey} /></div>
      </div>

      <div className="column is-1" />
      <div className="column is-11">
        <p>
          {content}
        </p>
      </div>
    </Fragment>
  }
  </Fragment>
);

const ParticipantList = ({ participations }) => (
  <div>
    {participations && participations.map((participation, index) => {
      const coloration = index % 2 === 0 ? 'has-background-black has-text-white' : 'has-background-white has-text-black';
      return (
        <div className="columns" key={`participantModal-${participation.userId}`}>
          <div className="column is-3 is-mobile has-text-right is-fixed-bottom commenter">{index + 1}. {participation.firstName} {participation.lastName}</div>
          {participation.comment &&
          <Fragment>
            <div className="column is-9 is-mobile is-paddingless">
              <div className={`speech-bubble ${coloration}`}>
                {participation.comment}
              </div>
            </div>
          </Fragment>
        }
        </div>
      );
    })}
  </div>
);

EventModal.propTypes = {
  eventId: PropTypes.string,
  events: PropTypes.object,
  closeModal: PropTypes.func,
  participations: PropTypes.object,
  categories: PropTypes.object,
  settings: PropTypes.object,
};

ParticipantList.propTypes = {
  participations: PropTypes.array,
};

ModalItem.propTypes = {
  translationKey: PropTypes.string,
  content: PropTypes.string,
};
