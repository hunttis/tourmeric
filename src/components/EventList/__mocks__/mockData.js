export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
