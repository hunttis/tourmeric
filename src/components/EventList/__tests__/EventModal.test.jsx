import React from 'react';
import { shallow } from 'enzyme';
import EventModal from '../EventModal';
import { mockEventId, mockParticipations, mockSingleEvent } from '../__mocks__/mockData';

describe('EventModal tests', () => {

  it('Compares EventModal to snapshot', () => {

    const closeModalFunction = () => {};

    const modal = shallow(<EventModal
      eventId={mockEventId}
      closeModal={closeModalFunction}
    />);

    expect(modal).toMatchSnapshot();
  });
});
