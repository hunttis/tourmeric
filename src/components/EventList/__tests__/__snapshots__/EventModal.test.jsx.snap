// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventModal tests Compares EventModal to snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventModal
    closeModal={[Function]}
    eventContent={
      Object {
        "category": "category1",
        "createDate": "2018-05-24T17:00:46.500Z",
        "date": "2018-09-24",
        "entryFee": "10",
        "format": "Miniatures",
        "link": "http://www.google.fi",
        "name": "Warhammer thing",
        "notes": "Bring your war face!",
        "playerSlots": "10",
        "prizes": "Glory!",
        "published": true,
        "rulesLevel": "Casual",
        "time": "Kill everything",
      }
    }
    eventId="mockEventId1"
    participations={
      Array [
        Object {
          "mockEventId1": Object {
            "mockUserId": Object {
              "comment": "Jee",
              "date": "2018-06-23T09:40:26.992Z",
              "firstName": "Test",
              "lastName": "Person",
              "userId": "mockUserId",
            },
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "modalmockEventId1",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="modal-background"
          onClick={[Function]}
        />,
        <div
          className="modal-content"
        >
          <div
            className="box is-rounded"
          >
            <div
              className="columns is-multiline"
            >
              <ModalItem
                content="Bring your war face!"
                translationKey="notes"
              />
              <ModalItem
                content="Glory!"
                translationKey="prizes"
              />
              <div
                className="column is-12"
              >
                <div
                  className="title"
                >
                  <Translate
                    id="link"
                  />
                </div>
                <p>
                  <a
                    href="http://www.google.fi"
                  >
                    http://www.google.fi
                  </a>
                </p>
              </div>
              <div
                className="column is-12"
              >
                <div>
                  <div
                    className="title"
                  >
                    <Translate
                      id="participants"
                    />
                  </div>
                  <ParticipantList
                    participations={
                      Array [
                        Object {
                          "mockEventId1": Object {
                            "mockUserId": Object {
                              "comment": "Jee",
                              "date": "2018-06-23T09:40:26.992Z",
                              "firstName": "Test",
                              "lastName": "Person",
                              "userId": "mockUserId",
                            },
                          },
                        },
                      ]
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>,
        <button
          aria-label="close"
          className="modal-close is-large"
          onClick={[Function]}
        />,
      ],
      "className": "modal",
      "id": "modalmockEventId1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "modal-background",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="box is-rounded"
          >
            <div
              className="columns is-multiline"
            >
              <ModalItem
                content="Bring your war face!"
                translationKey="notes"
              />
              <ModalItem
                content="Glory!"
                translationKey="prizes"
              />
              <div
                className="column is-12"
              >
                <div
                  className="title"
                >
                  <Translate
                    id="link"
                  />
                </div>
                <p>
                  <a
                    href="http://www.google.fi"
                  >
                    http://www.google.fi
                  </a>
                </p>
              </div>
              <div
                className="column is-12"
              >
                <div>
                  <div
                    className="title"
                  >
                    <Translate
                      id="participants"
                    />
                  </div>
                  <ParticipantList
                    participations={
                      Array [
                        Object {
                          "mockEventId1": Object {
                            "mockUserId": Object {
                              "comment": "Jee",
                              "date": "2018-06-23T09:40:26.992Z",
                              "firstName": "Test",
                              "lastName": "Person",
                              "userId": "mockUserId",
                            },
                          },
                        },
                      ]
                    }
                  />
                </div>
              </div>
            </div>
          </div>,
          "className": "modal-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="columns is-multiline"
            >
              <ModalItem
                content="Bring your war face!"
                translationKey="notes"
              />
              <ModalItem
                content="Glory!"
                translationKey="prizes"
              />
              <div
                className="column is-12"
              >
                <div
                  className="title"
                >
                  <Translate
                    id="link"
                  />
                </div>
                <p>
                  <a
                    href="http://www.google.fi"
                  >
                    http://www.google.fi
                  </a>
                </p>
              </div>
              <div
                className="column is-12"
              >
                <div>
                  <div
                    className="title"
                  >
                    <Translate
                      id="participants"
                    />
                  </div>
                  <ParticipantList
                    participations={
                      Array [
                        Object {
                          "mockEventId1": Object {
                            "mockUserId": Object {
                              "comment": "Jee",
                              "date": "2018-06-23T09:40:26.992Z",
                              "firstName": "Test",
                              "lastName": "Person",
                              "userId": "mockUserId",
                            },
                          },
                        },
                      ]
                    }
                  />
                </div>
              </div>
            </div>,
            "className": "box is-rounded",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ModalItem
                  content="Bring your war face!"
                  translationKey="notes"
                />,
                <ModalItem
                  content="Glory!"
                  translationKey="prizes"
                />,
                <div
                  className="column is-12"
                >
                  <div
                    className="title"
                  >
                    <Translate
                      id="link"
                    />
                  </div>
                  <p>
                    <a
                      href="http://www.google.fi"
                    >
                      http://www.google.fi
                    </a>
                  </p>
                </div>,
                <div
                  className="column is-12"
                >
                  <div>
                    <div
                      className="title"
                    >
                      <Translate
                        id="participants"
                      />
                    </div>
                    <ParticipantList
                      participations={
                        Array [
                          Object {
                            "mockEventId1": Object {
                              "mockUserId": Object {
                                "comment": "Jee",
                                "date": "2018-06-23T09:40:26.992Z",
                                "firstName": "Test",
                                "lastName": "Person",
                                "userId": "mockUserId",
                              },
                            },
                          },
                        ]
                      }
                    />
                  </div>
                </div>,
              ],
              "className": "columns is-multiline",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": "Bring your war face!",
                  "translationKey": "notes",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": "Glory!",
                  "translationKey": "prizes",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="title"
                    >
                      <Translate
                        id="link"
                      />
                    </div>,
                    <p>
                      <a
                        href="http://www.google.fi"
                      >
                        http://www.google.fi
                      </a>
                    </p>,
                  ],
                  "className": "column is-12",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Translate
                        id="link"
                      />,
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "id": "link",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href="http://www.google.fi"
                      >
                        http://www.google.fi
                      </a>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "http://www.google.fi",
                        "href": "http://www.google.fi",
                      },
                      "ref": null,
                      "rendered": "http://www.google.fi",
                      "type": "a",
                    },
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div>
                    <div
                      className="title"
                    >
                      <Translate
                        id="participants"
                      />
                    </div>
                    <ParticipantList
                      participations={
                        Array [
                          Object {
                            "mockEventId1": Object {
                              "mockUserId": Object {
                                "comment": "Jee",
                                "date": "2018-06-23T09:40:26.992Z",
                                "firstName": "Test",
                                "lastName": "Person",
                                "userId": "mockUserId",
                              },
                            },
                          },
                        ]
                      }
                    />
                  </div>,
                  "className": "column is-12",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="title"
                      >
                        <Translate
                          id="participants"
                        />
                      </div>,
                      <ParticipantList
                        participations={
                          Array [
                            Object {
                              "mockEventId1": Object {
                                "mockUserId": Object {
                                  "comment": "Jee",
                                  "date": "2018-06-23T09:40:26.992Z",
                                  "firstName": "Test",
                                  "lastName": "Person",
                                  "userId": "mockUserId",
                                },
                              },
                            },
                          ]
                        }
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Translate
                          id="participants"
                        />,
                        "className": "title",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "id": "participants",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "participations": Array [
                          Object {
                            "mockEventId1": Object {
                              "mockUserId": Object {
                                "comment": "Jee",
                                "date": "2018-06-23T09:40:26.992Z",
                                "firstName": "Test",
                                "lastName": "Person",
                                "userId": "mockUserId",
                              },
                            },
                          },
                        ],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "close",
          "className": "modal-close is-large",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "modalmockEventId1",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal-background"
            onClick={[Function]}
          />,
          <div
            className="modal-content"
          >
            <div
              className="box is-rounded"
            >
              <div
                className="columns is-multiline"
              >
                <ModalItem
                  content="Bring your war face!"
                  translationKey="notes"
                />
                <ModalItem
                  content="Glory!"
                  translationKey="prizes"
                />
                <div
                  className="column is-12"
                >
                  <div
                    className="title"
                  >
                    <Translate
                      id="link"
                    />
                  </div>
                  <p>
                    <a
                      href="http://www.google.fi"
                    >
                      http://www.google.fi
                    </a>
                  </p>
                </div>
                <div
                  className="column is-12"
                >
                  <div>
                    <div
                      className="title"
                    >
                      <Translate
                        id="participants"
                      />
                    </div>
                    <ParticipantList
                      participations={
                        Array [
                          Object {
                            "mockEventId1": Object {
                              "mockUserId": Object {
                                "comment": "Jee",
                                "date": "2018-06-23T09:40:26.992Z",
                                "firstName": "Test",
                                "lastName": "Person",
                                "userId": "mockUserId",
                              },
                            },
                          },
                        ]
                      }
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>,
          <button
            aria-label="close"
            className="modal-close is-large"
            onClick={[Function]}
          />,
        ],
        "className": "modal",
        "id": "modalmockEventId1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "modal-background",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="box is-rounded"
            >
              <div
                className="columns is-multiline"
              >
                <ModalItem
                  content="Bring your war face!"
                  translationKey="notes"
                />
                <ModalItem
                  content="Glory!"
                  translationKey="prizes"
                />
                <div
                  className="column is-12"
                >
                  <div
                    className="title"
                  >
                    <Translate
                      id="link"
                    />
                  </div>
                  <p>
                    <a
                      href="http://www.google.fi"
                    >
                      http://www.google.fi
                    </a>
                  </p>
                </div>
                <div
                  className="column is-12"
                >
                  <div>
                    <div
                      className="title"
                    >
                      <Translate
                        id="participants"
                      />
                    </div>
                    <ParticipantList
                      participations={
                        Array [
                          Object {
                            "mockEventId1": Object {
                              "mockUserId": Object {
                                "comment": "Jee",
                                "date": "2018-06-23T09:40:26.992Z",
                                "firstName": "Test",
                                "lastName": "Person",
                                "userId": "mockUserId",
                              },
                            },
                          },
                        ]
                      }
                    />
                  </div>
                </div>
              </div>
            </div>,
            "className": "modal-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="columns is-multiline"
              >
                <ModalItem
                  content="Bring your war face!"
                  translationKey="notes"
                />
                <ModalItem
                  content="Glory!"
                  translationKey="prizes"
                />
                <div
                  className="column is-12"
                >
                  <div
                    className="title"
                  >
                    <Translate
                      id="link"
                    />
                  </div>
                  <p>
                    <a
                      href="http://www.google.fi"
                    >
                      http://www.google.fi
                    </a>
                  </p>
                </div>
                <div
                  className="column is-12"
                >
                  <div>
                    <div
                      className="title"
                    >
                      <Translate
                        id="participants"
                      />
                    </div>
                    <ParticipantList
                      participations={
                        Array [
                          Object {
                            "mockEventId1": Object {
                              "mockUserId": Object {
                                "comment": "Jee",
                                "date": "2018-06-23T09:40:26.992Z",
                                "firstName": "Test",
                                "lastName": "Person",
                                "userId": "mockUserId",
                              },
                            },
                          },
                        ]
                      }
                    />
                  </div>
                </div>
              </div>,
              "className": "box is-rounded",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ModalItem
                    content="Bring your war face!"
                    translationKey="notes"
                  />,
                  <ModalItem
                    content="Glory!"
                    translationKey="prizes"
                  />,
                  <div
                    className="column is-12"
                  >
                    <div
                      className="title"
                    >
                      <Translate
                        id="link"
                      />
                    </div>
                    <p>
                      <a
                        href="http://www.google.fi"
                      >
                        http://www.google.fi
                      </a>
                    </p>
                  </div>,
                  <div
                    className="column is-12"
                  >
                    <div>
                      <div
                        className="title"
                      >
                        <Translate
                          id="participants"
                        />
                      </div>
                      <ParticipantList
                        participations={
                          Array [
                            Object {
                              "mockEventId1": Object {
                                "mockUserId": Object {
                                  "comment": "Jee",
                                  "date": "2018-06-23T09:40:26.992Z",
                                  "firstName": "Test",
                                  "lastName": "Person",
                                  "userId": "mockUserId",
                                },
                              },
                            },
                          ]
                        }
                      />
                    </div>
                  </div>,
                ],
                "className": "columns is-multiline",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "content": "Bring your war face!",
                    "translationKey": "notes",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "content": "Glory!",
                    "translationKey": "prizes",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="title"
                      >
                        <Translate
                          id="link"
                        />
                      </div>,
                      <p>
                        <a
                          href="http://www.google.fi"
                        >
                          http://www.google.fi
                        </a>
                      </p>,
                    ],
                    "className": "column is-12",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Translate
                          id="link"
                        />,
                        "className": "title",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "id": "link",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href="http://www.google.fi"
                        >
                          http://www.google.fi
                        </a>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "http://www.google.fi",
                          "href": "http://www.google.fi",
                        },
                        "ref": null,
                        "rendered": "http://www.google.fi",
                        "type": "a",
                      },
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div>
                      <div
                        className="title"
                      >
                        <Translate
                          id="participants"
                        />
                      </div>
                      <ParticipantList
                        participations={
                          Array [
                            Object {
                              "mockEventId1": Object {
                                "mockUserId": Object {
                                  "comment": "Jee",
                                  "date": "2018-06-23T09:40:26.992Z",
                                  "firstName": "Test",
                                  "lastName": "Person",
                                  "userId": "mockUserId",
                                },
                              },
                            },
                          ]
                        }
                      />
                    </div>,
                    "className": "column is-12",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <div
                          className="title"
                        >
                          <Translate
                            id="participants"
                          />
                        </div>,
                        <ParticipantList
                          participations={
                            Array [
                              Object {
                                "mockEventId1": Object {
                                  "mockUserId": Object {
                                    "comment": "Jee",
                                    "date": "2018-06-23T09:40:26.992Z",
                                    "firstName": "Test",
                                    "lastName": "Person",
                                    "userId": "mockUserId",
                                  },
                                },
                              },
                            ]
                          }
                        />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <Translate
                            id="participants"
                          />,
                          "className": "title",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "id": "participants",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "type": "div",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "participations": Array [
                            Object {
                              "mockEventId1": Object {
                                "mockUserId": Object {
                                  "comment": "Jee",
                                  "date": "2018-06-23T09:40:26.992Z",
                                  "firstName": "Test",
                                  "lastName": "Person",
                                  "userId": "mockUserId",
                                },
                              },
                            },
                          ],
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "close",
            "className": "modal-close is-large",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
